test/functorch/test_control_flow.py

```diff
@@ -1210,7 +1216,104 @@ def foo(x):
 
         gm = make_fx(foo, tracing_mode="symbolic")(torch.ones(3, 2, 1))
         x = torch.ones(4, 3, 2)
-        self.assertEqual(foo(x), gm(x))
+        self.assertEqual(gm(x), true_fn(x))
+        self.assertEqual(foo(x), true_fn(x))
+
+    def _check_closure_correcly_lifted(self, f, *, args, exp_res, exp_arg_num):
+        assert isinstance(args, (tuple, list))
+        self.assertEqual(f(*args), exp_res)
+        gm = make_fx(f)(*args)
+        self.assertEqual(gm(*args), exp_res)
+
+        def cnt_placeholder(gm):
+            return len([node for node in gm.graph.nodes if node.op == "placeholder"])
+        placeholder_cnts = [cnt_placeholder(mod) for mod in gm.children()]
+        self.assertTrue(all(cnt == exp_arg_num for cnt in placeholder_cnts))
+
+    def test_cond_with_tensor_closure(self):
+        a = torch.ones(2, 3)
+        b = torch.ones(2, 3) + 1
+
+        def true_fn(x):
+            return x + a
+
+        def false_fn(x):
+            return x + b
+
+        def foo(x):
+            return cond(x.shape[0] == 4, true_fn, false_fn, [x])
+
+
+        inp = torch.ones(2, 3)
+        res = foo(inp)
```

Comment:
I want to make sure that cond is still correct when a/b change. Could we add some tests where we do foo(inp), then modify a and b, and then call foo(inp) again?